core/src/main/scala/org/apache/spark/scheduler/BarrierJobAllocationFailed.scala

```diff
@@ -33,7 +33,9 @@ private[spark] class BarrierJobRunWithDynamicAllocationException
   extends BarrierJobAllocationFailed(
     BarrierJobAllocationFailed.ERROR_MESSAGE_RUN_BARRIER_WITH_DYN_ALLOCATION)
 
-private[spark] class BarrierJobSlotsNumberCheckFailed
+private[spark] class BarrierJobSlotsNumberCheckFailed(
```

Comment:
Would DAGScheduler not already have these values? maybe not, just checking whether it must be passed back in the message.

Reply:
I think it's difficult to know from DAGScheduler which barrier stage in a job requires more slots. So a stage which requires more slots need to tell that to DAGScheduler by passing back that in the message.